File: features/lead/src/main/java/com/nivasafinance/features/leadlender/enums/LeadLenderStatus.java
package com.nivasafinance.features.leadlender.enums;

public enum LeadLenderStatus {
    SELECTED,
    SUBMITTED,
    REJECTED;

    public boolean isInProgress() {
        return this == SELECTED || this == SUBMITTED;
    }

    public boolean isCompletedOrRejected() {
        return this == REJECTED;
    }
}

File: features/lead/src/main/java/com/nivasafinance/features/leadlender/enum/LeadLenderStatus.kt
package com.nivasafinance.features.leadlender.enum

enum class LeadLenderStatus {
    PROPOSED,
    LOGGED_IN,
    REJECTED,
    ;

    val isInProgress: Boolean
        get() = this == PROPOSED || this == LOGGED_IN
    val isCompletedOrRejected: Boolean
        get() = this == REJECTED
}
